flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1139,17 +1138,9 @@ public void triggerCheckpointBarrier(
 
 		if (executionState == ExecutionState.RUNNING && invokable != null) {
 
-			// build a local closure
-			final SafetyNetCloseableRegistry safetyNetCloseableRegistry =
-				FileSystemSafetyNet.getSafetyNetCloseableRegistryForThread();
-
 			Runnable runnable = new Runnable() {
 				@Override
 				public void run() {
-					// set safety net from the task's context for checkpointing thread
```

Comment:
> Are you sure we can remove this? What was the purpose of this safety net?

Do you have any reasons to keep it?

For the second part of your comment, I don't think it would be correct to allow any other (non main task thread) to get letters from the mailbox. And so far, we haven't raised any reason to have such requirement (that letters can be run by some other threads).

Reply:
> Do you have any reasons to keep it?

I don't know this code and I'm not sure how it's suppose to work, but I see that it's being used (`FileSystemSafetyNet#initializeSafetyNetForThread/closeSafetyNetAndGuardedResourcesForThread/wrapWithSafetyNetWhenActivated`). Since you were removing it, I thought that you have investigated how is it suppose to be working and that's safe to remove it (as I'm pretty sure that such safety feature has a very poor test coverage).

> For the second part of your comment, I don't think it would be correct to allow any other (non main task thread) to get letters from the mailbox. And so far, we haven't raised any reason to have such requirement (that letters can be run by some other threads).

I guess that's right as long as we do not allow for yielding in the legacy source threads. Do we have a `checkstate` somewhere for that? Somewhere Inside yield a `checkState(isMailboxThread())`?

Reply:
My assumption that we are not calling any user code and file system operations in the dispatcher thread anymore. Hence, we can remove the `FileSystemSafetyNet` guard usage there.
I'm open for corrections, if someone finds this to be not true.

> Do we have a checkstate somewhere for that? Somewhere Inside yield a checkState(isMailboxThread())?

There are checks that you can find in the code (and you should have read it from the preceding PRs).

Reply:
To me it looks like `FileSystemSafetyNet.setSafetyNetCloseableRegistryForThread` should be called by any thread that will be executing some/any part of the code that used to be invoked in `invokable.triggerCheckpoint`. This is the entry point, where safety net registry was being set, and then it was being used implicitly via static variables in `SnapshotDirectory` via `FileSystem#getLocalFileSystem`. 

> we are not calling any user code and file system operations in the dispatcher thread anymore.

I guess yes, probably setting it here now doesn't make sense, but it shouldn't be just removed from here, but moved used inside the mailbox action?

Reply:
Mailbox action is executed by the main task's thread. I think that thread has already `FileSystemSafetyNet` guard set.